dot/parachain/prospective-parachains/prospective-parachains.go

```diff
@@ -0,0 +1,81 @@
+package prospective_parachains
+
+import (
+	"context"
+
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/internal/log"
+)
+
+var logger = log.NewFromGlobal(log.AddContext("pkg", "prospective_parachains"), log.SetLevel(log.Debug))
+
+type ProspectiveParachains struct {
+	SubsystemToOverseer chan<- any
+}
+
+// Name returns the name of the subsystem
+func (*ProspectiveParachains) Name() parachaintypes.SubSystemName {
+	return parachaintypes.ProspectiveParachains
+}
+
+// NewProspectiveParachains creates a new ProspectiveParachain subsystem
+func NewProspectiveParachains(overseerChan chan<- any) *ProspectiveParachains {
+	prospectiveParachain := ProspectiveParachains{
+		SubsystemToOverseer: overseerChan,
+	}
+	return &prospectiveParachain
+}
+
+// Run starts the CandidateValidation subsystem
+func (pp *ProspectiveParachains) Run(ctx context.Context, overseerToSubsystem <-chan any) {
+	for {
+		select {
+		case msg := <-overseerToSubsystem:
+			pp.processMessage(msg)
+		case <-ctx.Done():
+			if err := ctx.Err(); err != nil {
```

Comment:
I guess on node shutdown, the caller of `Run()` cancels the context. Maybe we should add a check for that here to avoid logging errors on clean shutdown.

```suggestion
			if err := ctx.Err(); err != nil && !errors.Is(err, context.Canceled) {
```

Reply:
Updated.